### PR TITLE
Cleanup CPROVER Manual

### PR DESCRIPTION
**Pull request by tautschnig**
This fixes the following inconsistencies:
- Update numbering in index.md to avoid duplicate indices.
- Fix anchors in on the goto-analyzer page.
- Add links to JBMC, contracts to make sure the information in
doc/cprover-manual is actually visible on the rendered page.
- Replace the blank page that clicking on "variants" in the goto-cc
section produced by the content of the file that it was meant to link
to.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: doc/cprover-manual/api.md [hunk 1]
```diff
@@ -164,7 +164,7 @@ Uninterpreted functions are documented [here](./modeling-nondeterminism.md)).
 ### Memory-Related Functions
 
 The semantics of the primitives listed in this section is described in more detail in the
-document about [Memory Primitives](http://cprover.diffblue.com/memory-primitives.html).
+document about [Memory Primitives](./memory-primitives.md).
```
Comment (kroening) [inline doc/cprover-manual/api.md]: Does linking to a `.md` file work?

File: doc/cprover-manual/index.md [hunk 2]
```diff
@@ -10,23 +10,28 @@
 [Loop Unwinding](cbmc/unwinding/),
 [Assertion Checking](cbmc/assertions/)
 
-## 4. [Goto-Analyzer &ndash; Abstract Interpretation](goto-analyzer/)
+## 4. [Compositional Reasoning using Code Contracts](contracts/)
 
-## 4. [Test Suite Generation](test-suite/)
+## 5. [JBMC &ndash; Bounded Model Checking Java Bytecode](jbmc-user-manual/)
```
Comment (kroening) [inline doc/cprover-manual/index.md]: As mentioned, I think JBMC is so different that it makes sense to have a separate user manual.

File: doc/cprover-manual/api.md (hunk 1 above)
Comment (tautschnig) [reply doc/cprover-manual/api.md]: Fair point, I had seen this done in api.md, but looking at the rendered result this does not seem to work. I'll use try links via the TOC names instead (and will fix those in api.md).

File: doc/cprover-manual/index.md (hunk 2 above)
Comment (tautschnig) [reply doc/cprover-manual/index.md]: Yes, I just wondered where to actually put it... I'll remove it for now.

File: doc/cprover-manual/api.md (hunk 1 above)
Comment (tautschnig) [reply doc/cprover-manual/api.md]: Now done.

File: doc/cprover-manual/cbmc-tutorial.md [hunk 3]
```diff
@@ -138,11 +138,11 @@ possible output format.
 ```
 
 The specification of the XML trace output can be found here: [XML
```
Comment (kroening) [inline doc/cprover-manual/cbmc-tutorial.md]: A tutorial should absolutely not have a pointer to the XML spec.

File: doc/cprover-manual/index.md [hunk 4]
```diff
@@ -10,23 +10,26 @@
...
-## 6. Modeling
+## 7. [Program Properties](properties/)
+
+## 8. Modeling
 
 [Nondeterminism](modeling/nondeterminism/),
 [Assumptions](modeling/assumptions/),
 [Pointers](modeling/pointers/),
 [Floating Point](modeling/floating-point/)
+[File-local Functions](static-functions/)
```
Comment (kroening) [inline doc/cprover-manual/index.md]: Please don't add. There needs to be a proper section on modular verification. This is very, very confusing.

Comment (tautschnig) [reply doc/cprover-manual/index.md]: Will remove, but as with the JBMC part I do wonder where that place shall be.

File: doc/cprover-manual/cbmc-tutorial.md (hunk 3 above)
Comment (tautschnig) [reply doc/cprover-manual/cbmc-tutorial.md]: Done (as a second commit).

File: doc/cprover-manual/index.md (hunk 4 above)
Comment (kroening) [reply doc/cprover-manual/index.md]: It could become part of a new section on modular verification, which has contract-based verification as a subsection.

**Review by kroening — APPROVED** [review]

File: doc/cprover-manual/index.md [hunk 5]
```diff
@@ -10,23 +10,25 @@
...
+## 8. Modeling
 
 [Nondeterminism](modeling/nondeterminism/),
 [Assumptions](modeling/assumptions/),
 [Pointers](modeling/pointers/),
 [Floating Point](modeling/floating-point/)
+[Generating Environments](goto-harness/)
 
-## 7. Build Systems
+## 8. Build Systems
```
Comment (feliperodri) [inline doc/cprover-manual/index.md]: This should be 9.

**Review by feliperodri — APPROVED** [review]
Thanks for this update, @tautschnig! Only a minor comment.

File: doc/cprover-manual/goto-analyzer.md [hunk 6]
```diff
@@ -59,7 +59,7 @@ goto-analyzer --show --dot depgraph.dot --dependence-graph-vs
 
 
 
-### Task {#task}
+### <a name="task"></a> Task
```
Comment (SaswatPadhi) [inline doc/cprover-manual/goto-analyzer.md]: The `name` attribute has been deprecated in HTML5. See: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-name

`id` should be used instead, although the `id` is generally auto-generated (from the heading content) by both GitHub's markdown viewer, and all major Markdown -> HTML converters.

**Review by SaswatPadhi — COMMENTED** [review]
Thanks for cleaning this up!

One small comment, but not a blocker.

File: doc/cprover-manual/goto-analyzer.md (hunk 6 above)
Comment (tautschnig) [reply doc/cprover-manual/goto-analyzer.md]: Thank you for the clarification and also making me take a second look at what is being rendered right now (which has broken links). You are absolutely right that `id` is auto-generated, it was just being confused by the extra part in braces (we ended up with `id="tasktask"`).

**Pull request merged**